packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,474 @@
+
+### Tooltips in v8/Fabric
+
+v8 tooltips use a `TooltipHost` wrapped around the target element to provide tooltip functionality. This creates a `div` around the element that listens for mouse and focus events.
+
+The `Tooltip` component renders as a `Callout`, and supports all `Callout` props.
+
+```jsx
+<TooltipHost
+  content="This is the tooltip content"
+  // This id is used on the tooltip itself, not the host
+  // (so an element with this id only exists when the tooltip is shown)
+  id={tooltipId}
+  calloutProps={calloutProps}
+  styles={hostStyles}
+>
+  <DefaultButton aria-describedby={tooltipId}>Hover over me</DefaultButton>
+</TooltipHost>
+```
+
+#### Drawbacks
+
+There are a few drawbacks with this approach to adding tooltips, which are outlined in [☂ Tooltip: open issues to resolve in converged approach #15102](https://github.com/microsoft/fluentui/issues/15102), and summarized below:
+
+- The wrapper `div` created by `TooltipHost` can cause layout issues for the component. It also doesn't always result in proper positioning for the tooltip.
+- The API is overly complex.
+- It's not possible to have a tooltip within a tooltip
+- There's no coordination between tooltips on a page. For example, moving the mouse between two elements with tooltips should cause the second tooltip to appear immediately without fading in/out.
+
+### Tooltips in v0/Northstar
+
+v0 tooltips use a `trigger` property to render the tooltip's target component. However, unlike v8 it does not create a wrapper `div` around the target component, but instead adds listeners to the target component's props.
+
+```jsx
+<Tooltip content="Example tooltip" trigger={<Button content="A button" />} />
+```
+
+# Sample Code
+
+There are a few ways to add a tooltip to a component.
+
+## A component with a `tooltip` slot
+
+Components can choose to expose a pseudo-slot `tooltip`, by inheriting props from the `WithTooltipSlot` interface and using the `useTooltipSlot` hook.
+
+While this looks and acts like a normal slot to users, the `Tooltip` is not actually rendered by the component, but rather passed to the `TooltipManager` to render. The `useTooltipSlot` hook is lightweight (small bundle size impact); it does not have a direct dependency on the `Tooltip` or `TooltipManager` classes.
+
+For example, to add the `tooltip` slot to Button:
+
+```typescript
+// Button.types.ts:
+export interface ButtonProps extends WithTooltipSlot /*...*/ {
+  // ...
+}
+
+// useButton.ts:
+export const useButton = (/*...*/) => {
+  const state = mergeProps(/*...*/);
+
+  // ...
+  useTooltipSlot(state);
+  // ...
+
+  return state;
+};
+```
+
+Example usage:
+
+```jsx
+<Button tooltip="Example tooltip" />
+<Button tooltip={<>Custom <b>Tooltip</b> content!</>} />
+<Button tooltip={{ children: 'Placed Tooltip', placement: 'right' }} />
+<Button tooltip={{ as: MyTooltip, children: 'You can even implement your own tooltip' }} />
+```
+
+## On any element using a ref
```

Comment:
don't we need third way as well - that would not require having a ref, but would just add the tooltip to any rendered component? Ideally however, API of tooltip and popup should be similar

Reply:
Not all built components consumed from libraries (internal or OSS) are guaranteed to forward refs, so sometimes using a ref just might be possible

Reply:
I replaced useTooltipRef with a wrapper "TooltipTrigger" component. It won't add any DOM elements (i.e. no wrapper <div>), but it'll just attach listeners to its child component.